### PR TITLE
Fix `this` duplicate serialization in snapshot

### PR DESCRIPTION
**Pull request by jpbempel**
# What Does This Do
Remove duplicate serialization of `this`

# Motivation
`this` instance was serialized twice by instrumentation adding this as
 argument and by explicit serialization in json adapters.
This duplicate was deduplicated automatically when parsed on the backend side

# Additional Notes
Fix unit tests that were incorrect and cannot use directly fields anymore

### CODE REVIEW

File: dd-java-agent/agent-debugger/debugger-bootstrap/src/main/java/datadog/trace/bootstrap/debugger/ProbeRateLimiter.java [hunk 1]
```diff
@@ -52,6 +52,11 @@ public static void resetGlobalRate() {
     setGlobalSnapshotRate(DEFAULT_GLOBAL_LOG_RATE);
   }
 
+  public static void resetAll() {
```
Comment (ojung) [inline dd-java-agent/agent-debugger/debugger-bootstrap/src/main/java/datadog/trace/bootstrap/debugger/ProbeRateLimiter.java:55]: Is this related to the PR?

File: dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/agent/SnapshotSerializationTest.java [hunk 2]
```diff
@@ -96,18 +97,18 @@ public void roundTripCapturedValue() throws IOException, URISyntaxException {
     JsonAdapter<Snapshot> adapter = createSnapshotAdapter();
     Snapshot snapshot = createSnapshot();
     CapturedContext context = new CapturedContext();
-    CapturedContext.CapturedValue normalValuedField =
-        CapturedContext.CapturedValue.of("normalValuedField", String.class.getTypeName(), "foobar");
-    CapturedContext.CapturedValue normalNullField =
-        CapturedContext.CapturedValue.of("normalNullField", String.class.getTypeName(), null);
+    CapturedContext.CapturedValue normalValuedLocal =
```
Comment (ojung) [inline dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/agent/SnapshotSerializationTest.java:100]: Why are you changing fields to locals here and below?

**Review by ojung — APPROVED** [review]
Nice, good job fixing this 😂

File: dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/agent/SnapshotSerializationTest.java (hunk 2 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/agent/SnapshotSerializationTest.java:100]: because I cannot put directly fields like before need to generate a this argument to put fields, which more cumbersome, does not change the point of the test in this case
